The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/12407882884/job/34638627711) [(plain)](https://github.com/rust-lang/rust/commit/72c89fbabf0ace5db1d841cf951878ce5a4bc89a/checks/34638627711/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain

COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---
#13 3.158 Building wheels for collected packages: reuse
#13 3.159   Building wheel for reuse (pyproject.toml): started
#13 3.409   Building wheel for reuse (pyproject.toml): finished with status 'done'
#13 3.409   Created wheel for reuse: filename=reuse-4.0.3-cp310-cp310-manylinux_2_35_x86_64.whl size=132720 sha256=026f3bb0f1aa8090b861fd0a0939cb1a782396d84c8aab7875096557d637a0f6
#13 3.410   Stored in directory: /tmp/pip-ephem-wheel-cache-qn7ikecq/wheels/3d/8d/0a/e0fc6aba4494b28a967ab5eaf951c121d9c677958714e34532
#13 3.412 Installing collected packages: boolean-py, binaryornot, tomlkit, reuse, python-debian, markupsafe, license-expression, jinja2, chardet, attrs
#13 3.799 Successfully installed attrs-23.2.0 binaryornot-0.4.4 boolean-py-4.0 chardet-5.2.0 jinja2-3.1.4 license-expression-30.3.0 markupsafe-2.1.5 python-debian-0.1.49 reuse-4.0.3 tomlkit-0.13.0
#13 3.799 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#13 4.350 Collecting virtualenv
#13 4.350 Collecting virtualenv
#13 4.409   Downloading virtualenv-20.28.0-py3-none-any.whl (4.3 MB)
#13 4.682      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.3/4.3 MB 15.9 MB/s eta 0:00:00
#13 4.743 Collecting platformdirs<5,>=3.9.1
#13 4.754   Downloading platformdirs-4.3.6-py3-none-any.whl (18 kB)
#13 4.798 Collecting filelock<4,>=3.12.2
#13 4.832 Collecting distlib<1,>=0.3.7
#13 4.842   Downloading distlib-0.3.9-py2.py3-none-any.whl (468 kB)
#13 4.859      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 469.0/469.0 KB 31.2 MB/s eta 0:00:00
#13 4.859      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 469.0/469.0 KB 31.2 MB/s eta 0:00:00
#13 4.940 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#13 5.124 Successfully installed distlib-0.3.9 filelock-3.16.1 platformdirs-4.3.6 virtualenv-20.28.0
#13 DONE 5.2s

#14 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#14 DONE 0.0s
---
DirectMap4k:      108480 kB
DirectMap2M:     8280064 kB
DirectMap1G:    10485760 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
    Finished `dev` profile [unoptimized] target(s) in 0.04s
##[endgroup]
WARN: currently no CI rustc builds have rustc debug assertions enabled. Please either set `rust.debug-assertions` to `false` if you want to use download CI rustc or set `rust.download-rustc` to `false`.
downloading https://static.rust-lang.org/dist/2024-11-27/rustfmt-nightly-x86_64-unknown-linux-gnu.tar.xz
---
     ///
     /// The raw pointer must point to a block of memory allocated by the global allocator
-    /// 
+    ///
     /// The user of `from_raw` has to make sure a specific value of `T` is only
     /// dropped once.
     ///
fmt error: Running `"/checkout/obj/build/x86_64-unknown-linux-gnu/rustfmt/bin/rustfmt" "--config-path" "/checkout" "--edition" "2021" "--unstable-features" "--skip-children" "--check" "/checkout/library/alloc/src/collections/vec_deque/into_iter.rs" "/checkout/library/alloc/src/collections/vec_deque/drain.rs" "/checkout/library/alloc/src/collections/vec_deque/spec_from_iter.rs" "/checkout/library/alloc/src/collections/vec_deque/macros.rs" "/checkout/library/alloc/src/collections/vec_deque/tests.rs" "/checkout/library/alloc/src/collections/vec_deque/spec_extend.rs" "/checkout/library/alloc/src/collections/vec_deque/mod.rs" "/checkout/library/alloc/src/collections/vec_deque/iter.rs" "/checkout/library/alloc/src/collections/vec_deque/iter_mut.rs" "/checkout/library/alloc/src/collections/mod.rs" "/checkout/library/alloc/src/collections/btree/fix.rs" "/checkout/library/alloc/src/collections/btree/append.rs" "/checkout/library/alloc/src/collections/btree/navigate.rs" "/checkout/library/alloc/src/collections/btree/set/entry.rs" "/checkout/library/alloc/src/collections/btree/set/tests.rs" "/checkout/library/alloc/src/collections/btree/set_val.rs" "/checkout/library/alloc/src/collections/btree/node.rs" "/checkout/library/alloc/src/collections/btree/split.rs" "/checkout/library/alloc/src/collections/btree/remove.rs" "/checkout/library/alloc/src/collections/btree/node/tests.rs" "/checkout/library/alloc/src/collections/btree/borrow.rs" "/checkout/library/alloc/src/collections/btree/borrow/tests.rs" "/checkout/library/alloc/src/collections/btree/map.rs" "/checkout/library/alloc/src/collections/btree/merge_iter.rs" "/checkout/library/alloc/src/collections/btree/map/entry.rs" "/checkout/library/alloc/src/collections/btree/map/tests.rs" "/checkout/library/alloc/src/collections/btree/dedup_sorted_iter.rs" "/checkout/library/alloc/src/collections/btree/mod.rs" "/checkout/library/alloc/src/collections/btree/set.rs" "/checkout/library/alloc/src/collections/btree/mem.rs" "/checkout/library/alloc/src/collections/btree/search.rs" "/checkout/library/alloc/src/borrow.rs" "/checkout/library/alloc/src/fmt.rs" "/checkout/library/alloc/src/str.rs" "/checkout/library/alloc/src/rc/tests.rs" "/checkout/library/alloc/src/raw_vec.rs" "/checkout/library/alloc/src/slice.rs" "/checkout/library/alloc/src/lib.rs" "/checkout/library/alloc/src/testing/rng.rs" "/checkout/library/alloc/src/testing/ord_chaos.rs" "/checkout/library/alloc/src/testing/crash_test.rs" "/checkout/library/alloc/src/testing/mod.rs" "/checkout/library/alloc/src/macros.rs" "/checkout/library/alloc/src/sync.rs" "/checkout/library/alloc/src/task.rs" "/checkout/library/alloc/src/vec/splice.rs" "/checkout/library/alloc/src/vec/extract_if.rs" "/checkout/library/alloc/src/vec/into_iter.rs" "/checkout/library/alloc/src/vec/cow.rs" "/checkout/library/alloc/src/vec/drain.rs" "/checkout/library/alloc/src/vec/in_place_drop.rs" "/checkout/library/alloc/src/vec/in_place_collect.rs" "/checkout/library/alloc/src/vec/partial_eq.rs" "/checkout/library/alloc/src/vec/spec_from_iter.rs" "/checkout/library/alloc/src/vec/set_len_on_drop.rs" "/checkout/library/alloc/src/vec/spec_from_iter_nested.rs" "/checkout/library/alloc/src/vec/spec_from_elem.rs" "/checkout/library/alloc/src/vec/is_zero.rs" "/checkout/library/alloc/src/vec/spec_extend.rs" "/checkout/library/alloc/src/vec/mod.rs" "/checkout/library/alloc/benches/binary_heap.rs" "/checkout/library/alloc/benches/string.rs" "/checkout/library/alloc/benches/linked_list.rs" "/checkout/library/alloc/src/collections/binary_heap/mod.rs"` failed.
If you're running `tidy`, try again with `--bless`. Or, if you just want to format code, run `./x.py fmt` instead.
  local time: Thu Dec 19 07:13:36 UTC 2024
  network time: Thu, 19 Dec 2024 07:13:36 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/library/alloc/src/sync.rs
+++ b/library/alloc/src/sync.rs
@@ -1376,6 +1376,8 @@ impl<T: ?Sized> Arc<T> {
     /// different types. See [`mem::transmute`][transmute] for more information
     /// on what restrictions apply in this case.
     ///
+    /// The raw pointer must point to a block of memory allocated by the global allocator
+    ///
     /// The user of `from_raw` has to make sure a specific value of `T` is only
     /// dropped once.
     ///
@@ -1431,7 +1433,8 @@ impl<T: ?Sized> Arc<T> {
     ///
     /// The pointer must have been obtained through `Arc::into_raw`, and the
     /// associated `Arc` instance must be valid (i.e. the strong count must be at
-    /// least 1) for the duration of this method.
+    /// least 1) for the duration of this method, and `ptr` must point to a block of memory
+    /// allocated by the global allocator.
     ///
     /// # Examples
     ///
@@ -1465,7 +1468,8 @@ impl<T: ?Sized> Arc<T> {
     ///
     /// The pointer must have been obtained through `Arc::into_raw`, and the
     /// associated `Arc` instance must be valid (i.e. the strong count must be at
-    /// least 1) when invoking this method. This method can be used to release the final
+    /// least 1) when invoking this method, and `ptr` must point to a block of memory
+    /// allocated by the global allocator. This method can be used to release the final
     /// `Arc` and backing storage, but **should not** be called after the final `Arc` has been
     /// released.
     ///

